include/alpaka/mem/buf/Traits.hpp

```diff
@@ -99,7 +99,8 @@ namespace alpaka
                 typename TElem,
                 typename TDim,
                 typename TIdx>
-            using Buf = typename traits::BufType<TDev, TElem, TDim, TIdx>::type;
+            using Buf = typename traits::BufType<
+                alpaka::dev::Dev<TDev>, TElem, TDim, TIdx>::type;
```

Comment:
I like this version. You are passing in something (`TDev`) that can give you the real device type via `alpaka::dev::Dev<>`. This extends the valid types easily.
If it is already a device, it uses the reflexive `alpaka::dev::Dev<>` specialization of the device type itself.
If it is not a device, it simply has to have a specialization of `alpaka::dev::Dev<>`.